### ISSUE TITLE
Use existing Datalab to audit additional new data (e.g. test set)

### ISSUE THREAD

**Issue by jwmueller** [cleanlab/cleanlab#820]
Current Datalab is specific to one dataset only. 
After the Datalab has been used to audit the original dataset, we want the fitted Datalab to be able to audit extra test data that is supposedly from the same underlying distribution. 

Here the audit of test data from the existing Datalab should **re-use statistics** computed from the original dataset (rather than re-computing them on the new dataset).  

The audit of additional test data can look for new types of issues as well, which are specific to discrepancies between the new data and original data:

- out of distribution examples in the new data that not similar to any of the original data
- new types of label issues in the new data (eg. new classes)
- drift between the new data and original data
- higher-loss between labels and predictions in the new data vs original data.

Important: We need a tutorial that makes it clear when it is preferable to do either of the following:

Option 1:  (can be done with existing functionality)

```
all_data = merge(original_data, new_data)
lab = Datalab(...) 
lab.find_issues(all_data, ...)
```

Option 2: (requires the new functionality to be built as specified in this GH issue)

```
lab = Datalab(...)
lab.find_issues(original_data, ...)

lab_new = Datalab(train_datalab = lab, ...)
lab_new.find_issues(new_data, ...)
```

The answer is Option 2 is preferable in settings where:
1. new data is streaming in quickly (and perhaps in small batches)
2. we assume new data to have same distribution as original data, and are specifically interested in issues that capture various failures of this assumption (OOD examples, new classes in new_data, etc.)

`lab_new.find_issues()` should not assume access to the original dataset, only things that have been stored in the original Datalab `lab`. This function should run fairly quickly when the `new_data` is small, regardless if `original_data` was large.

**Comment by jwmueller** [comment]
If working on this, the starting point is: 

1. extend Datalab to be optionally instantiated from an existing Datalab (that was fitted to the original data aka training data)
2. first issue type to work on is detecting OOD examples in the new data (measure their KNN distance to the old data when new data features are provided), as demonstrated in: 
https://docs.cleanlab.ai/stable/tutorials/outliers.html

**Comment by ChChCh8** [comment]
I'd like to work on this!

**Comment by coding-famer** [comment]
I'd like to work on this!

**Comment by jwmueller** [comment]
@coding-famer Amazing! Please sync with @elisno about details before commencing the implementation. This is something where figuring out the exact spec before coding will speed things up greatly.
